django/db/models/base.py

```diff
@@ -1169,6 +1204,15 @@ def _do_insert(self, manager, using, fields, returning_fields, raw):
             raw=raw,
         )
 
+    def _assign_returned_values(self, returned_values, returning_fields):
+        returning_fields_iter = iter(returning_fields)
+        for value, field in zip(returned_values, returning_fields_iter):
+            setattr(self, field.attname, value)
+        # Defer all fields that were meant to be updated with their database
+        # resolved values but couldn't as they are effectively stale.
+        for field in returning_fields_iter:
+            self.__dict__.pop(field.attname, None)
```

Comment:
Marking the fields as deferred on backends that don't support `RETURNNG` seems like a smarter choice than keeping them around in their expression form (as [that can cause unexpected issues](https://docs.djangoproject.com/en/5.2/ref/models/expressions/#f-assignments-persist-after-model-save)) but more importantly as it would otherwise incur an extra database query on each save that might be unnecessary in most cases.